internal/analysis/analysis_mode.go

```diff
@@ -0,0 +1,25 @@
+package analysis
+
+// Mode is used to distinguish between whether static or dynamic analysis is being performed
+type Mode string
+
+const (
+	Dynamic     Mode = "dynamic"
+	Static      Mode = "static"
+	InvalidMode Mode = ""
+)
+
+func AllModes() []Mode {
+	return []Mode{Dynamic, Static}
+}
+
+func ModeFromString(s string) Mode {
```

Comment:
Rather than a sentinel value, prefer e.g. 

```go
func ModeFromString(s string) (Mode, bool) {
  
} 

mode, ok := ModeFromString(s)
...
```

In this case, because the error case is very simple, a bool suffices. If error cases become more complicated, then an `error` would be more appropriate.